libraries/entities/src/index.js

```diff
@@ -23,7 +23,26 @@ class Entity {
 
   create(properties, overwrite) {
     const documentId = this.calculateId(properties);
-    this.db.procs[this.serviceName][`${this.tableName}_create`](documentId, properties, overwrite, 1);
+
+    return this.db.procs[`${this.tableName}_create`](documentId, properties, overwrite, 1);
+  }
+
+  delete(properties) {
+    const documentId = this.calculateId(properties);
+
+    return this.db.procs[`${this.tableName}_delete`](documentId);
+  }
+
+  update(properties) {
+    const documentId = this.calculateId(properties);
+
+    return this.db.procs[`${this.tableName}_update`](documentId, properties, 1);
+  }
+
+  load(properties) {
+    const documentId = this.calculateId(properties);
+
+    return this.db.procs[`${this.tableName}_load`](documentId);
   }
 
```

Comment:
Questions:

1. If a user tries to update an entry that doesn't exist, where do we throw the error? I wasn't sure if we should handle errors here or inside the stored procedure in `0001.yml`.
2. Should we rely on the implementation of the stored procedure to update the etag when an entry is modified (current behavior) or is this something we should take care of inside the entity class?

Reply:
1. The errors will need some specific JS behavior (`err.code = "ResourceNotFound"`).  So the stored procedure should return enough information for the JS to construct an appropriate error.
2. It seems like it would be easiest to do so in the stored procedure, since you get to check within a transaction.  We'll probably need to use some locking or the like to ensure serializability.